src/PowerShellEditorServices/Services/TextDocument/Handlers/CodeLensHandlers.cs

```diff
@@ -40,12 +39,17 @@ protected override CodeLensRegistrationOptions CreateRegistrationOptions(CodeLen
 
         public override Task<CodeLensContainer> Handle(CodeLensParams request, CancellationToken cancellationToken)
         {
+            _logger.LogDebug($"Handling code lens request for {request.TextDocument.Uri}");
+
             ScriptFile scriptFile = _workspaceService.GetFile(request.TextDocument.Uri);
-            CodeLens[] codeLensResults = ProvideCodeLenses(scriptFile, cancellationToken);
-            return Task.FromResult(new CodeLensContainer(codeLensResults));
+            IEnumerable<CodeLens> codeLensResults = ProvideCodeLenses(scriptFile, cancellationToken);
+
+            return !codeLensResults.Any()
```

Comment:
`ProvideCodeLenses` returns a yield enumerator right? If so then the call to `Any` will double enumerate the first item at least (meaning it'll double invoke the method basically). Best to just pass the enumerator along, not worth the potential allocation save

Side note, we should probably look at a way to make this actually async. I know this is already the case, but the method being entirely synchronous while pretending to be async is a little concerning